client/components/platform-checkout/save-user/phone-number-input.js

```diff
@@ -1,144 +1,111 @@
 import utils from 'iti/utils';
 
-const PhoneNumberInput = ( { handlePhoneNumberChange } ) => {
-	const [ inputValue, setInputValue ] = useState(
-		document.getElementById( 'billing_phone' )?.value ?? ''
-	);
+const PhoneNumberInput = ( {
+	onValueChange,
+	value,
+	onValidationChange = () => {},
+	inputProps = {},
+} ) => {
 	const [ inputInstance, setInputInstance ] = useState( null );
 	const [ isValid, setIsValid ] = useState( true );
+	const inputRef = useRef();
 
-	const handlePhoneNumberInputChange = ( e ) => {
-		setInputValue( e.target.value );
+	const handlePhoneNumberInputChange = () => {
 		if ( inputInstance ) {
-			handlePhoneNumberChange( inputInstance.getNumber() );
+			onValueChange( inputInstance.getNumber() );
+			onValidationChange( inputInstance.isValidNumber() );
 		}
 	};
 
-	const handlePhoneNumberValidation = () => {
+	const removeInternationalPrefix = ( phone ) => {
 		if ( inputInstance ) {
-			setIsValid( inputInstance.isValidNumber() );
-		} else {
-			setIsValid( true );
-		}
-	};
-
-	const ErrorText = () => {
-		if ( ! isValid ) {
-			const message = __(
-				'Please enter a valid mobile phone number.',
-				'woocommerce-payments'
+			return phone.replace(
+				'+' + inputInstance.getSelectedCountryData().dialCode,
+				''
 			);
-
-			return <p className="error-text">{ message }</p>;
 		}
-		return '';
-	};
-
-	useEffect( () => {
-		// const formSubmitButton = document.getElementById( 'place_order' );
-		const formSubmitButton = document.querySelector(
-			'form.woocommerce-checkout button[type="submit"]'
-		);
-
-		if ( ! formSubmitButton ) {
-			return;
-		}
-
-		const updateFormSubmitButton = () => {
-			if ( isValid ) {
-				formSubmitButton.removeAttribute( 'disabled' );
-			} else {
-				formSubmitButton.setAttribute( 'disabled', 'disabled' );
-			}
-		};
-
-		updateFormSubmitButton();
 
-		return () => {
-			// Clean up
-			formSubmitButton.removeAttribute( 'disabled' );
-		};
-	}, [ isValid ] );
+		return value;
+	};
 
 	useEffect( () => {
 		let iti = null;
-		const input = document.querySelector(
-			`input[aria-label="${ __(
-				'Mobile phone number',
-				'woocommerce-payments'
-			) }"]`
-		);
 
 		const handleCountryChange = () => {
-			handlePhoneNumberChange( iti.getNumber() );
+			onValueChange( iti.getNumber() );
 		};
 
-		if ( input ) {
-			iti = intlTelInput( input, {
+		if ( inputRef.current ) {
+			iti = intlTelInput( inputRef.current, {
 				initialCountry: 'US',
 				customPlaceholder: () => '',
 				separateDialCode: true,
 				hiddenInput: 'full',
 				utilsScript: utils,
 			} );
 			setInputInstance( iti );
-			setInputValue( ( currentInputValue ) =>
-				currentInputValue.replace(
-					'+' + iti.getSelectedCountryData().dialCode,
-					''
-				)
-			);
+			setIsValid( iti.isValidNumber() );
+			onValidationChange( iti.isValidNumber() );
 
 			// Focus the phone number input when the component loads.
-			input.focus();
+			inputRef.current.focus();
```

Comment:
I'm not sure why this is used. I believe we shouldn't automatically focus on the mobile input when the page is loaded, but maybe I'm missing something...

Should I just go and remove it?